
date: 2002/10/30 17:36:30;  author: jer;  state: Exp;  lines: +1 -1
lame bug, fix double-connects from services and the chaos that ensues

Index: jabberd/base/base_accept.c
===================================================================
RCS file: /home/cvs/jabberd14/jabberd/base/base_accept.c,v
retrieving revision 1.66
retrieving revision 1.67
diff -u -r1.66 -r1.67
--- jabberd/base/base_accept.c	8 Feb 2002 07:44:14 -0000	1.66
+++ jabberd/base/base_accept.c	30 Oct 2002 17:36:30 -0000	1.67
@@ -144,7 +144,7 @@
 
         case MIO_XML_NODE:
             /* If aio has been authenticated previously, go ahead and deliver the packet */
-            if(ai->state == A_READY)
+            if(ai->state == A_READY  && m == ai->m)
             {
                 /* Hide 1.0 style transports etherx:* attribs */
                 xmlnode_hide_attrib(x, "etherx:to");
